services/rsources/extension.go

```diff
@@ -0,0 +1,130 @@
+package rsources
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/rudderlabs/rudder-server/config"
+)
+
+type extension interface {
+	getReadDB() *sql.DB
+	createStatsTable(ctx context.Context, jobRunId string) error
+	dropTables(ctx context.Context, jobRunId string) error
+	cleanupLoop(ctx context.Context) error
+}
+
+type defaultExtension struct {
+	localDB *sql.DB
+}
+
+func (r *defaultExtension) getReadDB() *sql.DB {
+	return r.localDB
+}
+
+func (r *defaultExtension) createStatsTable(ctx context.Context, jobRunId string) error {
+	dbHost := config.GetEnv("JOBS_DB_HOST", "localhost")
+	jobRunIdStatTableName := fmt.Sprintf("%s%s", tablePrefix, jobRunId)
```

Comment:
What is the cardinality of `jobRunId` ?
How many tables do we expect to have at any given point in time?

Reply:
https://rudderlabs.slack.com/archives/C017H2XKZ9R/p1653390164299659